test/integration/offchain/demand.test.ts

```diff
@@ -0,0 +1,302 @@
+import { describe, before } from 'mocha'
+import { Express } from 'express'
+import { expect } from 'chai'
+
+import createHttpServer from '../../../src/server'
+import { post, get } from '../../helper/routeHelper'
+import {
+  seed,
+  cleanup,
+  parametersAttachmentId,
+  exampleDate,
+  seededDemandAId,
+  seededDemandBId,
+  nonExistentId,
+  seededDemandACreationTransactionId,
+  seededDemandBCreationTransactionId,
+  seededDemandACreationTransactionId2,
+  seededDemandBCreationTransactionId2,
+  seededDemandACommentTransactionId,
+  seededDemandBCommentTransactionId,
+  seededDemandACommentTransactionId2,
+  seededDemandBCommentTransactionId2,
+  seededDemandAAlreadyAllocated,
+  seededDemandBAlreadyAllocated,
+} from '../../seeds'
+
+import { selfAlias, withIdentitySelfMock } from '../../helper/mock'
+import { assertIsoDate, assertUUID } from '../../helper/assertions'
+
+const runDemandTests = (demandType: 'demandA' | 'demandB') => {
+  const dbDemandSubtype = demandType === 'demandA' ? 'demand_a' : 'demand_b'
+  const seededDemandId = demandType === 'demandA' ? seededDemandAId : seededDemandBId
+  const seededCreationTransactionId =
+    demandType === 'demandA' ? seededDemandACreationTransactionId : seededDemandBCreationTransactionId
+  const seededCreationTransactionId2 =
+    demandType === 'demandA' ? seededDemandACreationTransactionId2 : seededDemandBCreationTransactionId2
+  const seededCommentTransactionId =
+    demandType === 'demandA' ? seededDemandACommentTransactionId : seededDemandBCommentTransactionId
+  const seededCommentTransactionId2 =
+    demandType === 'demandA' ? seededDemandACommentTransactionId2 : seededDemandBCommentTransactionId2
+  const seededDemandAlreadyAllocated =
+    demandType === 'demandA' ? seededDemandAAlreadyAllocated : seededDemandBAlreadyAllocated
```

Comment:
Yeah if you look at the top comment I mentioned this. It's really really messy and the best fix would be to split up the seed file so we could do this as a per test precondition rather than a monolithic seed. I would personally prefer to leave it like this as really this is a feature of the existing tests and not a regression. This just really highlights it in technicolor